source/common/network/raw_buffer_socket.cc

```diff
@@ -13,6 +13,7 @@ void RawBufferSocket::setTransportSocketCallbacks(TransportSocketCallbacks& call
 IoResult RawBufferSocket::doRead(Buffer::Instance& buffer) {
   PostIoAction action = PostIoAction::KeepOpen;
   uint64_t bytes_read = 0;
+  bool last_byte = false;
```

Comment:
This is not a fully formed comment, just a question: Do we have problems since we are using EV_CLOSED (FileReadyType::Closed)? Is there a case during flow control in which we might get EOF but not actually read all the data and do we now have to handle that? I know we had some issues with this in the past and don't enable EV_CLOSED in all cases, but I think it might cause problems.

Reply:
I actually suspect that if we are handling half close semantics we should never set EV_CLOSED?

Reply:
Yes, I think we will want to disable EV_CLOSED in this case.

Reply:
I think I just need to call detectEarlyCloseWhenReadDisabled(true) on both the upstream and downstream connections.  @alyssawilk does that seem correct?

Reply:
Yes I think that is correct. Basically if half close semantics are enabled never detect early close.